lib/repl.js

```diff
@@ -949,7 +950,30 @@ function REPLServer(prompt,
           self.displayPrompt();
           return;
         }
+      }
+
+      // In the next two if blocks, we do not use os.EOL instead of '\n'
+      // because on Windows it is '\r\n'
+      if (StringPrototypeIncludes(cmd, '\n')) { // If you are editing a multiline command
+        self.history[0] = self[kNormalizeHistoryLineEndings](cmd, '\n', '\r');
+      } else if (self[kBufferedCommandSymbol]) { // If a new multiline command was entered
+        // Remove the first N lines from the self.history array
+        // where N is the number of lines in the buffered command
+
+        const lines = StringPrototypeSplit(self[kBufferedCommandSymbol], '\n');
+        self.history = ArrayPrototypeSlice(self.history, lines.length);
+        lines[lines.length - 1] = cmd;
+        const newHistoryLine = ArrayPrototypeJoin(lines, '\r');
+        if (self.history[0] !== newHistoryLine) {
+          ArrayPrototypeUnshift(self.history, newHistoryLine);
+        }
+      }
+
+      if (e) {
         self._domain.emit('error', e.err || e);
+        self.lastCommandErrored = true;
+      } else {
+        self.lastCommandErrored = false;
```

Comment:
Should this state be moved to the top of the method? And we would need an internal symbol or it's public API.